classes/class3/README.md

```diff
@@ -1 +1,364 @@
+
+Now go ahead and copy the image `cat.jpg` from the `expressintro/public/images` folder in this directory and place it in your new images folder. Then navigate to http://localhost:3000/images/cat.jpg. That is your static folder doing work!
+
+## Templating
+
+Now I imagine you may be asking yourself, *"This is a web-dev course, right? Where's the HTML?"* Well its coming, but doing it smart is not as simple as putting some `<p>` tags in a file and calling it a day. We will be using a technique called templating. Templating allows us to put some logic behind what is otherwise entirely a layout language. This will greatly reduce the overall quantity of HTML you write, thereby reducing errors, and making it much easier to make a change to an element that appears on many pages in your app. Most importantly however, it allows us dynamically customize the contents of our HTML. Think about Facebook profiles. When you load yours, you are looking at a static page of HTML. But Facebook's servers aren't full of files called `ben-kahle-profile.html`, and `evan-simpson-profile.html` that wouldn't exactly scale very well. They have a single profile template into which they inject your personal profile information, render it as HTML and then send it to your browser. Let's learn the basics!
+
+First off, we will be using [Handlebars](http://handlebarsjs.com/) as a templating engine. The default templating engine of Express is actually [Jade](http://jade-lang.com/). We're going to use Handlebars for a few reasons:
+* It more closely resembles the HTML result you will see on the client. Some people consider this a negative, but we like it and it can make debugging your views *much* easier. It also makes it easier while you are learning HTML.
+* It supports pre-compiled templates and client side rendering. We won't get to these for a while, but they provide some great performance boosts and some awesome flexibility when it comes to designing and building dynamic front-ends.
+* Jade's whitespace dependency and style can be a bit too dense and harder to understand. Handlebars may be more "cluttered", but it can also be very nicely structured and layed out in an easy to read manner.
+
+At the end of the day, templating engines-like most things-come down to personal preference, and that's why we're forcing you to like Handlebars! (At least for the next couple weeks).
+
+### Making Express Render Handlebars
+
+To get Express to render handlebars, as well as to add some useful templating features which we will get into later, we are going to use the [Express-Handlebars](https://github.com/ericf/express-handlebars) module. So go ahead and `npm --save install express-handlebars` to get it installed.
+
+Now two more things to know about are layouts and partials.
+* Layouts: Basically meta-templates. They are handlebars documents that allow you to create a wrapper around all the templates that use them.
```

Comment:
I'm going to suggest swapping the example for partials up to layouts, as a nav bar is more layout-y. FB chat-like module as an example of a partial?